src/cost_functions.jl

```diff
@@ -54,17 +58,33 @@ function (f::L2Loss)(sol::DESolution)
       for j in 1:length(sol[i])
         sumsq +=(data[j,i] - sol[j,i])^2
       end
+      if diff_data != nothing
+        for j in 2:length(sol[i])
+          sumsq +=(diff_data[j-1,i] - sol[j,i] + sol[j-1,i])^2
```

Comment:
You don't actually need the `diff_data` for this one. Instead it's just `(data[j,i]-data[j,i-1] - sol[j,i] - sol[j,i-1])^2`. Notice the indexing by `i` too: the first column is through the solution at time point `i`.

Reply:
In that case, we always use the first difference too?

Reply:
We can do it based on whether there's a weight. But if the benchmarks show that it is good to use, it's probably always good to just do it then. The total cost should be minimal in comparison to solving the ODE itself, so anything that improves the estimate is good.

Reply:
Doing it always would probably be better